.github/workflows/cron-weekly-update-nightly.yml

```diff
@@ -0,0 +1,43 @@
+name: Update Nightly rustc
+on:
+  schedule:
+    - cron: "0 0 * * 0" # runs weekly on Sunday at 00:00
```

Comment:
I wanted to run a test job every day after nightly releases to help with uncovering bugs quickly but update the pin weekly sounds good.

Reply:
I'll make this daily and we'll see how annoying it is. In theory it's a one-ACK merge from me or you or @tcharding and 98% of the time no issues will appear.